src/util/migrate.cpp

```diff
@@ -108,6 +108,29 @@ type2tc migrate_type(const typet &type)
     array_type2t *a = new array_type2t(subtype, size, is_infinite);
     return type2tc(a);
   }
+  else if(type.id() == typet::t_vector)
+  {
+    type2tc subtype = migrate_type(type.subtype());
+    expr2tc size((expr2t *)nullptr);
+
+    if(type.find(typet::a_size).id() == "infinity")
+    {
+      assert(
+        0 &&
+        "Vector type has a constant size\n"
+        "Please, refer to: "
+        "https://clang.llvm.org/docs/"
+        "LanguageExtensions.html#vectors-and-extended-vectors");
+      abort();
```

Comment:
The assertion (well, the `if` above actually - for some reason the condition is not part of the assertion) checks an invariant the frontend is supposed to guarantee. Why is this `abort` needed here? It is internal and not an error caused by usage or environment.

Reply:
Replaced it with an assertion